libs/wingc/src/type_check.rs

```diff
@@ -4247,12 +4248,22 @@ impl<'a> TypeChecker<'a> {
 			self.spanned_error(variable, "Variable cannot be reassigned from inflight".to_string());
 		}
 
-		if matches!(kind, AssignmentKind::AssignIncr | AssignmentKind::AssignDecr) {
-			self.validate_type(exp_type, self.types.number(), value);
-			self.validate_type(var.type_, self.types.number(), variable);
+		// validate in case of increment/decrement reference assignament (e.g. `a+=1`)
+		match var.type_ {
+			_ if var.type_.is_subtype_of(&self.types.string()) && matches!(kind, AssignmentKind::AssignIncr) => {
+				self.validate_type(exp_type, self.types.string(), value);
+				self.validate_type(var.type_, self.types.string(), variable);
+			}
+			_ if var.type_.is_subtype_of(&self.types.number())
+				&& matches!(kind, AssignmentKind::AssignIncr | AssignmentKind::AssignDecr) =>
+			{
+				self.validate_type(exp_type, self.types.number(), value);
+				self.validate_type(var.type_, self.types.number(), variable);
+			}
+			_ => {
+				self.validate_type(exp_type, var.type_, value);
+			}
```

Comment:
The logic seems right, but using `match` here when everything is prefixed with a `_` seems cumbersome.
Try changing to a simple `if .. else if ..` chain:
```rust
if matches!(kind, AssignmentKind::AssignIncr) {
  self.validate_type_in(expr_type, &[self.types.number(), self.types.string()], value);
  self.validate_type_in(var.type_, &[self.types.number(), self.types.string()], value);
} else if matches!(kind, AssignmentKind::AssignDecr) {
  self.validate_type(expr_type, self.types.number(), value);
  self.validate_type(var.type_, self.types.number(), value);
}
```
or something similar

Reply:
I was on the same line before making this change (second commit); you are right, I'll rollback

Reply:
In addition, thanks for the validate_type_in; I didn't know it's existence